src/main/java/org/wso2/okta/client/OktaOAuthClient.java

```diff
@@ -0,0 +1,1046 @@
+            if (statusCode == HttpStatus.SC_NO_CONTENT) {
+                if (log.isDebugEnabled()) {
+                    log.debug(String.format("OAuth Client for the Consumer Key %s has been successfully deleted",
+                            clientId));
+                }
+            } else {
+                HttpEntity entity = response.getEntity();
+                reader = new BufferedReader(new InputStreamReader(entity.getContent(), OktaConstants.UTF_8));
+                JSONObject responseObject = getParsedObjectByReader(reader);
+                handleException(String.format("Problem occurred while deleting client for the Consumer Key %s." +
+                                " Response : %s", clientId, responseObject.toJSONString()));
+            }
+        } catch (IOException e) {
+            handleException("Error while reading response body from Server ", e);
+        } catch (ParseException e) {
+            handleException(OktaConstants.ERROR_WHILE_PARSE_RESPONSE, e);
+        } finally {
+            closeResources(reader, httpClient);
+        }
+    }
+
+    /**
+     * This method retrieves OAuth application details by given consumer key.
+     *
+     * @param clientId consumer key of the OAuth Client.
+     * @return an {@code OAuthApplicationInfo} having all the details of an OAuth Client.
+     * @throws APIManagementException This is the custom exception class for API management.
+     */
+    @Override
+    public OAuthApplicationInfo retrieveApplication(String clientId) throws APIManagementException {
+        if (log.isDebugEnabled()) {
+            log.debug(String.format("Retrieving an OAuth client from Okta authorization server for the Consumer Key: %s"
+                    , clientId));
+        }
+        // Getting Client Instance Url and API Key from Config.
+        String oktaInstanceUrl = configuration.getParameter(OktaConstants.OKTA_INSTANCE_URL);
+        String apiKey = configuration.getParameter(OktaConstants.REGISTRAION_API_KEY);
+        String registrationEndpoint = oktaInstanceUrl + OktaConstants.CLIENT_ENDPOINT;
+        if (StringUtils.isNotEmpty(clientId)) {
+            registrationEndpoint += "/" + clientId;
+        }
+
+        CloseableHttpClient httpClient = HttpClientBuilder.create().build();
+        BufferedReader reader = null;
+        try {
+            HttpGet request = new HttpGet(registrationEndpoint);
+            // Set authorization header, with API key.
+            request.addHeader(OktaConstants.AUTHORIZATION, OktaConstants.AUTHENTICATION_SSWS + apiKey);
+            if (log.isDebugEnabled()) {
+                log.debug(String.format("Invoking HTTP request to get the client details for the Consumer Key %s",
+                        clientId));
+            }
+            HttpResponse response = httpClient.execute(request);
+            int statusCode = response.getStatusLine().getStatusCode();
+            HttpEntity entity = response.getEntity();
```

Comment:
Do we need to check nullpointer in here.

Reply:
Fixed